velox/exec/SortWindowBuild.cpp

```diff
@@ -165,12 +165,36 @@ void SortWindowBuild::spill() {
   data_->pool()->release();
 }
 
-void SortWindowBuild::computePartitionStartRows() {
-  partitionStartRows_.reserve(numRows_);
+// Use double front and back search algorithm to find next partition start row.
+// it's better than linear or binary search.
+// Like to description of the algorithm:
+// https://medium.com/@insomniocode/search-algorithm-double-front-and-back-20f5f28512e7
+vector_size_t SortWindowBuild::findNextPartitionStartRow(vector_size_t start) {
   auto partitionCompare = [&](const char* lhs, const char* rhs) -> bool {
     return compareRowsWithKeys(lhs, rhs, partitionKeyInfo_);
   };
 
+  auto left = start;
+  auto right = left + 1;
+  auto lastPosition = sortedRows_.size();
+  while (right < lastPosition) {
```

Comment:
Is right here always left + 1?

Reply:
The initial value of right is left + 1.
In fact, right is calculated through right = left + distance.